File: SHMIP/shmip-hydro_intercomparison-06b3137c7049/input_functions/topography/python/sqrttopo.py
import numpy as np
def sqrttopo(x):
	''' Function defining the ice thickness for the SHMIP exercise.
	This is a square root shape on a flat bed
    
	Input:
	x coordinates (vector with the x coordinate of all points)
	Output:
	bed elevation and ice thickness in that order
	'''
	
	thick_para=6.
	thick_off=5.0e3
	min_thick=1.0
	
	#Geometry
	thickness=min_thick+thick_para*pow(x+thick_off,0.5)-6.0*pow(thick_off,0.5)
	base=np.zeros(np.shape(x))

	return base,thickness

File: SHMIP/shmip-hydro_intercomparison-06b3137c7049/input_functions/topography/matlab/sqrttopo.m
function [bed_ele, ice_thickness] = sqrttopo(x, y, bedslope)
% [bed_ele, ice_thickness] = sqrttopo(x, y, bedslope)
%
% Surface topography with square root shape and linearly sloping bed.
%
% Input:
% - slope: positive means increasing bed elevation

bed_ele = x*bedslope;

thick_para = 6;
thick_off = 5e3;
thick_nz = 1;
surf = thick_para*sqrt(x+thick_off) - thick_para*sqrt(thick_off) + thick_nz;
ice_thickness = surf-bed_ele;
